### ISSUE TITLE
Humanreadable gets monthly wrong

### ISSUE THREAD

**Issue by Ray-electrotechie** [rlanvin/php-rrule#116]
For the RRule:

```
'FREQ' => 'YEARLY',
'BYMONTH' => '4,5,6,7,8,9,10',
'COUNT' => '7',
'BYDAY' => '2SU',
'DTSTART' => '2023-01-01'
```

humanreadable returns:

> yearly on the second Sunday of the year in April, May, June, July, August, September and October, starting from 2023-01-01 00:00:00, 7 times

It should read:

> yearly on the second Sunday of the **month** in April, May, June, July, August, September and October, starting from 2023-01-01 00:00:00, 7 times

I am a rank amateur in php so I am incapable of helping to fix it.  I believe that I am using the latest version of rrule (cloned github and compared to the one I have). The code below is what I am using. I have NO idea how I can keep the line terminators in this system, so good luck reading it.

```
<?php
require __DIR__ . '/vendor/autoload.php';
require_once(__DIR__.'/vendor/rlanvin/php-rrule/src/RRule.php');
//RRULE:FREQ=YEARLY;BYMONTH=4,5,6,7,8,9,10;COUNT=6;BYDAY=2SU
$rrule2 = new RRule\RRule([
'FREQ' => 'YEARLY',
'BYMONTH' => '4,5,6,7,8,9,10',
'COUNT' => '7',
'BYDAY' => '2SU',
'DTSTART' => '2023-01-01'
]);
echo "RRULE:FREQ=YEARLY;BYMONTH=4,5,6,7,8,9,10;COUNT=7;BYDAY=2SU\n";
foreach ($rrule2 as $occurrence) {
    echo $occurrence->format('D d M Y'),", \n";
}
echo $rrule2->humanReadable(),"\n";
echo "It should read:\nyearly on the second Sunday of the month in April, May, June, July, August, September and October, starting from 2023-01-01 00:00`
```

```
php --version
PHP 7.4.3 (cli) (built: Aug 17 2022 13:29:56) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.3, Copyright (c), by Zend Technologies``
```

**Comment by rlanvin** [comment]
Hello, thanks for the report. I updated it to make it more readable.

It's an interesting case that I'm not sure how to fix (yet). I have been trying different approach for the human readable algorithms recently, so I'll try to incorporate that use case.

**Comment by Ray-electrotechie** [comment]
Thank you for taking the time to read my report and fix the formatting. I have programmed in a lot of languages, but I am a novice in php. I have been struggling to get rrules in microformatting within html so I gave up and decided to preprocess using your rrule software which is great. Google search seems focussed on the use of schema.org events, but that has a very poor event repetition facility - nothing like as good as rrule.  So my plan is to "flatten" the events and duplicate lots of info (per event) in the webpage.
Thanks for taking the time to produce and publish php rrule - much appreciated. I looked at your code and boy, is it complex.
